Lint error in `internals::slice::SliceExt`

```
error[E0658]: use of unstable library feature `slice_as_chunks`
 --> internals/src/slice.rs:14:18
```

Changing it to the below code works on nightly but then on stable it fails to compile due to the unstable feature:
```
    /// ```should_panic
    /// # #![feature(slice_as_chunks)]
    /// let slice = [1, 2, 3];
    /// let _fail = slice.as_chunks::<0>();
    /// ```
```

It would be possible to change the `compile_fail` tag to `ignore` instead of `should_panic`, but in the docs it looks better if the correct tag is used so that it gives the red highlight stating the example intentionally panics or fails to compile.

--- a/internals/src/slice.rs
+++ b/internals/src/slice.rs
@@ -9,10 +9,10 @@ pub trait SliceExt {
     ///
     /// Note that `N` must not be zero:
     ///
-    /// ```compile_fail
+    /// ```ignore
     /// # use bitcoin_internals::slice::SliceExt;
     /// let slice = [1, 2, 3];
-    /// let _fail = slice.bitcoin_as_chunks::<0>();
+    /// let _fail = slice.bitcoin_as_chunks::<0>(); // Fails to compile
     /// ```
     fn bitcoin_as_chunks<const N: usize>(&self) -> (&[[Self::Item; N]], &[Self::Item]);
 
@@ -20,10 +20,10 @@ pub trait SliceExt {
     ///
     /// Note that `N` must not be zero:
     ///
-    /// ```compile_fail
+    /// ```ignore
     /// # use bitcoin_internals::slice::SliceExt;
     /// let mut slice = [1, 2, 3];
-    /// let _fail = slice.bitcoin_as_chunks_mut::<0>();
+    /// let _fail = slice.bitcoin_as_chunks_mut::<0>(); // Fails to compile
     /// ```
     fn bitcoin_as_chunks_mut<const N: usize>(
         &mut self,

